src/java/org/apache/cassandra/service/reads/ReplicaFilteringProtection.java

```diff
@@ -137,8 +140,15 @@ public class ReplicaFilteringProtection<E extends Endpoints<E>>
 
         tableMetrics = ColumnFamilyStore.metricsFor(command.metadata().id);
 
-        this.cachedRowsWarnThreshold = cachedRowsWarnThreshold;
-        this.cachedRowsFailThreshold = cachedRowsFailThreshold;
+        if (System.getProperty(CACHED_REPLICA_ROWS_WARN_THRESHOLD.getKey()) != null)
+            this.cachedRowsWarnThreshold = Integer.getInteger(CACHED_REPLICA_ROWS_WARN_THRESHOLD.getKey());
+        else
+            this.cachedRowsWarnThreshold = cachedRowsWarnThreshold;
+
+        if (System.getProperty(CACHED_REPLICA_ROWS_FAIL_THRESHOLD.getKey()) != null)
+            this.cachedRowsFailThreshold = Integer.getInteger(CACHED_REPLICA_ROWS_FAIL_THRESHOLD.getKey());
+        else
+            this.cachedRowsFailThreshold = cachedRowsFailThreshold;
```

Comment:
I think this could be something like:
```
this.cachedRowsWarnThreshold = CACHED_REPLICA_ROWS_WARN_THRESHOLD.getInt(cachedRowsWarnThreshold);
this.cachedRowsFailThreshold = CACHED_REPLICA_ROWS_FAIL_THRESHOLD.getInt(cachedRowsFailThreshold);
```

This will make it easier to port to `main-5.0` where checkstyle will complain about using `System.getProperty`... :)

Reply:
Excellent suggestion :)